### ISSUE TITLE
id loss

### ISSUE THREAD

**Issue by hmmlencat** [idonahum/photoVerse#26]
why run_inference. use with torch.set_grad_enabled(training_mode and (i == len(scheduler.timesteps) - 1)),instead of with torch.no_grad().

**Comment by Haimzis** [comment]
Hi, when calculating the loss_id we do use gradients, the use of torch.no_grad() is only at logging step. 

I do agree that using torch.set_grad_enabled could help make the code more concise and shorter, and I appreciate your attention to this detail.
